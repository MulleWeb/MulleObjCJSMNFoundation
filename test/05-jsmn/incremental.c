//#define JSMN_STRICT
#include <MulleObjCJSMNFoundation/jsmn.h>
#include <stdio.h>
#include <string.h>

#define INCREMENTAL

static char  *json = "[\n"
"   1848,\n"
"   {\n"
"      \"key\": true\n"
"   }\n"
"]";


int main( void)
{
   jsmn_parser   p;
   jsmntok_t     t[128];
   int           r;
   int           i;
   size_t        len;

   jsmn_init( &p);

   len = strlen( json);

#ifdef INCREMENTAL
   for( i = 1; i <= len; i++)
   {
      r = jsmn_parse(&p, json, i, t, 128);
      if( r < 0)
      {
         if( r == JSMN_ERROR_PART)
            continue;
         fprintf( stderr, "Failed to parse JSON: %d\n", r);
         return 1;
      }
      break;
   }
#else
   r = jsmn_parse( &p, json, len, t, 128);
   if( r < 0)
   {
      fprintf( stderr, "Failed to parse JSON: %d\n", r);
      return 1;
   }
#endif
   for( i = 0; i < r; i++)
      printf( "%d: %.*s\n", i, t[ i].end - t[ i].start, &json[ t[ i].start]);
   return( 0);
}
