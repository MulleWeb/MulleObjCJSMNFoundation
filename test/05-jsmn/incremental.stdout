0: [
   1848,
   {
      "key": true
   }
]
1: 1848
2: {
      "key": true
   }
3: key
4: true
